packages/calypso-e2e/src/lib/components/sidebar-component.ts

```diff
@@ -25,4 +27,40 @@ export class SidebarComponent extends BaseContainer {
 	constructor( page: Page ) {
 		super( page, selectors.sidebar );
 	}
+
+	/**
+	 * Post-initialization steps of this object.
+	 *
+	 * @returns {Promise<void>} No return value.
+	 */
+	async _postInit(): Promise< void > {
+		await this.page.waitForLoadState( 'networkidle' );
+	}
+
+	/**
+	 * Clicks on the sidebar menu item matching the name.
+	 * Note that the menu item must be visible in some shape or form.
+	 * If there are multiple elements that match the selector, the first
+	 * matching element will be clicked (as per Playwright documentation).
+	 * Waits and assertions should be placed on the page/component that this
+	 * method will cause a navigation to, to ensure readiness before the script
+	 * continues.
+	 *
+	 * @param {string} name Plaintext name of the menu item in the sidebar.
+	 * @returns {Promise<void>} No return value.
+	 */
+	async clickMenuItem( name: string ): Promise< void > {
+		await this.page.click( `text=${ toTitleCase( name ) }` );
```

Comment:
Do you think it would be more efficient to specify the element selector as well? This is potentially querying all the elements on the page until finding the one that contains the text. I think it would also be easier to debug having the class/id selector at hand. 

For example:

```js
await this.page.click( `.menu-item:has-text( ${ toTitleCase( name ) } )` );
```

...or:

```js
await this.page.click( `.menu-item:text-matches( name, 'i' )` );
```

Reply:
I can't find the PR right now, but some time ago there was a flaky test (Selenium) where a selector with low specificity was causing the test to timeout because it was querying the document for too long (over 20s). AFAIR the selector was just `span` + some text to match. When I audited the page, I think over 1000 `span` elements were present (Gutenberg). I was able to fix that by simply making the selector more specific.

Reply:
I've filed #53556 to make the selector more specific. While so far it seems to work fine, you are right that more  specificity would not hurt here.

Reply:
I've started draft PR #53566 to address this, and it was quite a struggle surprisingly.